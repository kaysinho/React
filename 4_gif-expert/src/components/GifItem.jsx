import PropTypes from 'prop-types';

const GifItem = ( { img } ) => {
    return ( <div className="card"> <img src={ img.url } alt={ img.title } /> <p>{ img.title }</p> </div> );
}
 
export default GifItem;

GifItem.propTypes = {
    img: PropTypes.object.isRequired
}