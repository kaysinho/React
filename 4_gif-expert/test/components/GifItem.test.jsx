import { render, screen } from "@testing-library/react"
import GifItem from "../../src/components/GifItem"

describe('GifItemComponent', () => {

    const img = {
        img: {
            title: 'Example',
            url: 'https://example.com/image.jpg'
        }
    }


    test('Should be a match with Snapshot', () => {
        const { container } = render(<GifItem img={ img } />);
        expect(container).toMatchSnapshot()
    });

    test('Should render an Image passed', () => {
        render(<GifItem img={ img.img } />);
        //screen.debug()
        expect(screen.getByRole('img').src).toBe(img.img.url)
    });

    test('Should render an Alt passed', () => {
        render(<GifItem img={ img.img } />);
        //screen.debug()
        expect(screen.getByRole('img').alt).toBe(img.img.title)
    });

    test('Should render a Title', () => {
        render(<GifItem img={ img.img } />);
        //screen.debug()
        expect(screen.getByText(img.img.title)).toBeTruthy()
    });
})