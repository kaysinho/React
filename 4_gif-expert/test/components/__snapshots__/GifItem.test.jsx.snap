// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GifItemComponent Should be a match with Snapshot 1`] = `
<div>
  <div
    class="card"
  >
     
    <img />
     
    <p />
     
  </div>
</div>
`;
